### PR TITLE
Ddpb 1645

### PR DESCRIPTION
**Pull request by elvisciotti**

### CODE REVIEW

File: app/DoctrineMigrations/Version153.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() != 'postgresql', 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql('ALTER TABLE account DROP COLUMN IF EXISTS opening_date');
+        $this->addSql('ALTER TABLE account DROP COLUMN IF EXISTS closing_date');
+        $this->addSql('ALTER TABLE account DROP COLUMN IF EXISTS opening_date_explanation');
+        $this->addSql('ALTER TABLE account DROP COLUMN IF EXISTS closing_balance_explanation');
+        $this->addSql('ALTER TABLE account DROP COLUMN IF EXISTS closing_date_explanation');
```
Comment (slizzio) [inline app/DoctrineMigrations/Version153.php]: Is there a chance we'll lose some data with this?

File: src/AppBundle/Entity/Client.php [hunk 2]
```diff
@@ -44,14 +44,14 @@ class Client
      *
      * @JMS\Groups({"client-reports"})
      * @JMS\Type("array")
-     * @ORM\OneToMany(targetEntity="AppBundle\Entity\Report\Report", mappedBy="client", cascade={"persist"})
+     * @ORM\OneToMany(targetEntity="AppBundle\Entity\Report\Report", mappedBy="client", cascade={"persist", "remove"})
```
Comment (slizzio) [inline src/AppBundle/Entity/Client.php]: Dont we had a check for this within admin. Are we sure we want to do this? :)

File: src/AppBundle/Service/ReportService.php [hunk 3]
```diff
@@ -203,4 +205,44 @@ public function submit(Report $currentReport, User $user, \DateTime $submitDate)
...
+        }
+
+        $ret = [];
+        foreach($reports as $report1) {
+            foreach($reports as $report2) {
+                if ($report1->getId() === $report2->getId()) {
+                    continue;
+                }
+                // find report with same date that have not started
+                if ($report1->getStatus()->hasStarted()
```
Comment (slizzio) [inline src/AppBundle/Service/ReportService.php]: Should this be negated? comment says    // find report with same date that have not started

File: src/AppBundle/Service/ReportStatusService.php [hunk 4]
```diff
@@ -547,6 +547,22 @@ public function getSubmitState()
...
+     */
+    public function hasStarted()
+    {
+        $sectionStatus = $this->getSectionStatus();
+        // exclude balance, and money transfers, that depend on other section therefore not required
+        // to complete and therefore considered "done"
+        unset($sectionStatus['balance']);
+        unset($sectionStatus['moneyTransfers']);
+
+        return count(array_filter($sectionStatus, function ($e) {
```
Comment (slizzio) [inline src/AppBundle/Service/ReportStatusService.php]: typecast the count to (bool)

File: src/AppBundle/Entity/Client.php (hunk 2 above)
Comment (elvisciotti) [reply src/AppBundle/Entity/Client.php]: there is a db cascade delete  on client.report, and this is corresponding application-level cascade (only useful to keep the unity of work clean, as the cascade delete logic is already applied by the db). I needed this cascade to delete clients without users

File: app/DoctrineMigrations/Version153.php (hunk 1 above)
Comment (elvisciotti) [reply app/DoctrineMigrations/Version153.php]: columns have been removed lots of time ago, as the UX doesn't require nor used this data anymore. Very unlikely we need that, as old reports have been already submitted and it's not required for new reports

**Pull request merged**